modules/callbacks.inc

```diff
@@ -819,24 +819,40 @@ function entity_plus_metadata_taxonomy_access($op, $entity = NULL, $account = NU
 
 /**
  * Access callback for file entities.
+ * 
+ * @see entity_plus_access()
+ *
+ * @param string $op
+ *   The operation being performed. One of 'view', 'download', 'update', 'create' or
+ *   'delete'.
+ * @param File $file
+ *   A file to check access for. Must be a file object. Must have fid,
+ *   except in the case of 'create' operations.
+ * @param User $account
+ *   The user to check for. Leave it to NULL to check for the global user.
+ *
+ * @throws EntityMalformedException
+ *
+ * @return bool
+ *   TRUE if access is allowed, FALSE otherwise.
  */
-function entity_plus_metadata_file_access($op, $file = NULL, $account = NULL, $entity_plus_type = NULL) {
-  // We can only check access for the current user, so return FALSE on other accounts.
-  global $user;
-  if ($op == 'view' && isset($file) && (!isset($account) || $user->uid == $account->uid)) {
-    // Invoke hook_file_download() to obtain access information.
-    foreach (module_implements('file_download') as $module) {
-      $result = module_invoke($module, 'file_download', $file->uri);
-      if ($result == -1) {
-        return FALSE;
-      }
+function entity_plus_metadata_file_access($op, $file = NULL, $account = NULL) {
+  // First deal with the case where a $file is provided and op is 'create'.
+  if (isset($file) && $op == 'create') {
+    if (isset($file->type)) {
+      return file_access($op, $file->type, $account);
     }
+    else {
+      throw new EntityMalformedException('Permission to create a file was requested but no file type was given.');
```

Comment:
Should this text be wrapped in `t()`?

Reply:
I've checked adding a `t()` around the message and it works fine. I see no problem in adding it here and will do so next. 

There are other instances of `EntityMalformedException()` in Entity Plus that could also use `t()`. I'll open a separate issue for that.